pipelines/py/bump-constraints.yaml

```diff
@@ -0,0 +1,58 @@
+name: Bump Python constraint pins
+
+inputs:
+  constraints-file:
+    description: Path to the constraints file to update
+    default: constraints.txt
+  packages:
+    description: List of package==version pairs to update (newline separated, with optional comments)
+    required: true
+  only-replace:
+    description: Only update packages already in the constraints file (set to false to allow adding new packages)
+    default: "true"
+  show-diff:
+    description: Show diff of changes made
+    default: "false"
+
+needs:
+  packages:
+    - tw
+
+pipeline:
+  - name: "bump constraint pins"
+    runs: |
+      set -euo pipefail
+
+      constraints_file="${{inputs.constraints-file}}"
+      packages="${{inputs.packages}}"
+      only_replace="${{inputs.only-replace}}"
+      show_diff="${{inputs.show-diff}}"
+
+      # Check if constraints file exists
+      if [ ! -f "$constraints_file" ]; then
+        echo "ERROR: Constraints file '$constraints_file' not found" 1>&2
+        exit 1
+      fi
+
+      # Create a temporary file with the package updates
+      updates_file=$(mktemp /tmp/bump-constraints.XXXXXX)
+      trap "rm -f $updates_file" EXIT
+
+      # Write packages to the temporary file
+      echo "$packages" > "$updates_file"
+
+      # Prepare the only-replace flag
```

Comment:
i'm pretty sure you can just always pass `--only-replace=${{inputs.only-replace}}`.
that will save us ~5 lines of shell for the variable and the check here... let go do it.